### ISSUE TITLE
semigroups 3.0.12 release archive contains generated files

### ISSUE THREAD

**Issue by fingolfin** [gap-packages/Semigroups#453]
The semigroups 3.0.12 release tarball contains the following generated files, which I think it should rather not contain:

```
Makefile
bin/lib/libsemigroups.0.dylib
bin/lib/libsemigroups.a
bin/lib/libsemigroups.lai
config.log
config.status
libtool
src/.deps/semigroups_la-bipart.Plo
src/.deps/semigroups_la-congpairs.Plo
src/.deps/semigroups_la-converter.Plo
src/.deps/semigroups_la-fropin.Plo
src/.deps/semigroups_la-pkg.Plo
src/.deps/semigroups_la-semigrp.Plo
src/.deps/semigroups_la-uf.Plo
src/libsemigroups/Makefile
src/libsemigroups/config.log
src/libsemigroups/config.status
src/libsemigroups/libtool
src/libsemigroups/src/.deps/libsemigroups_la-blocks.Plo
src/libsemigroups/src/.deps/libsemigroups_la-cong.Plo
src/libsemigroups/src/.deps/libsemigroups_la-elements.Plo
src/libsemigroups/src/.deps/libsemigroups_la-rws.Plo
src/libsemigroups/src/.deps/libsemigroups_la-rwse.Plo
src/libsemigroups/src/.deps/libsemigroups_la-semigroups.Plo
src/libsemigroups/src/.deps/libsemigroups_la-uf.Plo
```

**Comment by james-d-mitchell** [comment]
Thanks @fingolfin, I'm not 100% sure how those got in there, I am obviously not cleaning the archive enough in the release script.

**Comment by fingolfin** [comment]
This is why in `ReleaseTools`, I create a complete fresh directory from scratch, which never was compiled; then there never were `.Plo` and `.la` etc. files in it, so no risk of accidentally copying them :-).

Out of curiosity: did you ever get further trying to use `ReleaseTools` for semigroups? If you are still interested in that, I'd be interested in helping to make it possible.

**Comment by james-d-mitchell** [comment]
I am using `ReleaseTools` for Semigroups, but a slightly hacked version which wraps libsemigroups in the release archive. The files above got included with the archive because currently to build the documentation requires loading the package which requires it to be built, and I wasn't cleaning the archive sufficiently after that (`make clean` instead of `make distclean && ./autogen.sh`). I will try to submit a PR to `ReleaseTools` but am currently a bit short of time.

**Comment by james-d-mitchell** [comment]
This is resolved in the version 3.0.13 release. Thanks @fingolfin
